3.3/test/run

```diff
@@ -152,8 +153,83 @@ for app in ${WEB_APPS[@]}; do
   test_connection
   check_result $?
 
-  info "All tests for the ${app}-test-app finished successfully."
+  info "All tests for the ${app} finished successfully."
   cleanup ${app}
 done
 
+
+app=unbuildable-test-app
+
+cid_file=$(mktemp -u --suffix=.cid)
+
+prepare ${app}
+run_sti_build ${app} 2>&1 | grep "'setup.py' not found" &> /dev/null && echo "Build failed as expected"
+check_result $?
+info "All tests for the ${app} finished successfully."
+cleanup ${app}
+
+
+app=django-test-apps/no-static
+
+cid_file=$(mktemp -u --suffix=.cid)
+
+prepare ${app}
+run_sti_build ${app} 2>&1 | grep -E "WARNING:.+ collectstatic" &> /dev/null && echo "Detected build warning as expected"
+check_result $?
+
+# Verify the 'usage' script is working properly when running the base image with 'sti usage ...'
+test_sti_usage
+check_result $?
+
+# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
+test_docker_run_usage
+check_result $?
+
+# Verify that the HTTP connection can be established to test application container
+run_test_application &
+
+# Wait for the container to write it's CID file
+wait_for_cid
+
+test_connection
+check_result $?
+
+info "All tests for the ${app} finished successfully."
+cleanup ${app}
+
+
+app=django-test-apps/with-static
+
+cid_file=$(mktemp -u --suffix=.cid)
+
+prepare ${app}
+run_sti_build ${app}
+check_result $?
+
+# Verify that static files are present
+STATIC_FILE=${test_dir}/${app}/staticfiles/admin/css/base.css
+[[ -f ${STATIC_FILE} ]] || echo "Could not find ${STATIC_FILE}"
+check_result $?
+
+# Verify the 'usage' script is working properly when running the base image with 'sti usage ...'
+test_sti_usage
+check_result $?
+
+# Verify the 'usage' script is working properly when running the base image with 'docker run ...'
+test_docker_run_usage
+check_result $?
+
+# Verify that the HTTP connection can be established to test application container
+run_test_application &
+
+# Wait for the container to write it's CID file
+wait_for_cid
+
+test_connection
+check_result $?
+
+info "All tests for the ${app} finished successfully."
+cleanup ${app}
+
+
```

Comment:
Instead of copy&pasting the contents of those 2 tests, wouldn't be better to run them in previous for loop, and add conditional check at the end of that for loop. I'm quite reluctant on accepting this solution.